stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/chat/navigation/GalleryImageAttachmentDestination.kt

```diff
@@ -1,31 +1,98 @@
 package io.getstream.chat.android.chat.navigation
 
+import android.app.Activity
 import android.content.Context
+import android.content.Intent
 import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.activity.result.launch
+import androidx.appcompat.app.AppCompatActivity
 import com.getstream.sdk.chat.model.ModelType
 import com.getstream.sdk.chat.navigation.destinations.AttachmentDestination
 import io.getstream.chat.android.client.models.Attachment
 import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.livedata.ChatDomain
 import io.getstream.chat.android.ui.gallery.AttachmentGalleryActivity
+import io.getstream.chat.android.ui.gallery.AttachmentGalleryActivity.Companion.EXTRA_ATTACHMENT_OPTION_RESULT
 import io.getstream.chat.android.ui.utils.extensions.getCreatedAtOrThrow
 
 public class GalleryImageAttachmentDestination(
     message: Message,
     attachment: Attachment,
     context: Context,
+    private val attachmentReplyOptionHandler: AttachmentGalleryActivity.AttachmentReplyOptionHandler,
+    private val attachmentShowInChatOptionHandler: AttachmentGalleryActivity.AttachmentShowInChatOptionHandler,
+    private val attachmentDownloadOptionHandler: AttachmentGalleryActivity.AttachmentDownloadOptionHandler,
+    private val onAttachmentDeleteOptionClickHandler: AttachmentGalleryActivity.AttachmentDeleteOptionHandler,
 ) : AttachmentDestination(message, attachment, context) {
     override fun showImageViewer(message: Message, attachment: Attachment) {
-        val attachments =
+        val attachments: List<Attachment> =
             message.attachments.filter { it.type == ModelType.attach_image && !it.imageUrl.isNullOrEmpty() }
 
         if (attachments.isEmpty()) {
             Toast.makeText(context, "Invalid image(s)!", Toast.LENGTH_SHORT).show()
             return
         }
 
-        val createdAt = message.getCreatedAtOrThrow().time
+        val createdAt: Long = message.getCreatedAtOrThrow().time
         val attachmentIndex = message.attachments.indexOf(attachment)
 
-        start(AttachmentGalleryActivity.createIntent(context, createdAt, attachmentIndex, message, attachments))
+        check(context is AppCompatActivity)
+        val launcher = context.activityResultRegistry.register(
```

Comment:
We discussed that it's too late to register for results here, I'll move it to a better point in the code

Reply:
Thanks for the help with this! Now handlers are not leaking and they service host activity death 👍🏼